### PR TITLE
feat: init @ovh-ux/ng-ovh-feature-flipping module 

### PR DESCRIPTION
**Pull request by Jisay**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-4540
| License          | BSD 3-Clause

## Description

A lib that will check features availabilities. It will request the 2API and expose the result.

Integration of this lib in SMS section.

## Ref:
- MANAGER-4543

### CODE REVIEW

File: packages/components/ng-ovh-feature-flipping/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# ng-ovh-feature-flipping
```
Comment (marie-j) [inline packages/components/ng-ovh-feature-flipping/README.md:1]: Is it global to ovh or just manager ?

**Comment by marie-j** [comment]
Is it possible and does it make sense instead maybe to have a ES6 lib and if needed a wrapper (for AngularJS in our case) ?

**Comment by Jisay** [comment]
> Is it possible and does it make sense instead maybe to have a ES6 lib and if needed a wrapper (for AngularJS in our case) ?

This is planned ;-)
For now, we will use as it in order to test the feature flipping mechanism.

File: packages/components/ng-ovh-feature-flipping/README.md (hunk 1 above)
Comment (Jisay) [reply packages/components/ng-ovh-feature-flipping/README.md:1]: This is global.
Rules in the 2API can be declared by order team and for order for example.

File: packages/components/ng-ovh-feature-flipping/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "prepare": "yarn run build",
+    "start": "lerna exec --stream --scope='@ovh-ux/ng-ovh-feature-flipping' --include-dependencies -- yarn run build",
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/ng-ovh-feature-flipping' --include-dependencies -- yarn run dev",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/ng-ovh-feature-flipping' --include-dependencies -- yarn run dev:watch"
+  },
+  "devDependencies": {
+    "@ovh-ux/component-rollup-config": "^7.0.0"
+  },
+  "peerDependencies": {
+    "angular": "~1.6"
```
Comment (marie-j) [inline packages/components/ng-ovh-feature-flipping/package.json:41]:
```suggestion
    "angular": "^1.7"
```
according to version used in manager

File: packages/components/ng-ovh-feature-flipping/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+];
+
+if (process.env.BUILD === 'production') {
+  outputs.push(config.cjs());
+  outputs.push(
+    config.umd({
+      output: {
+        globals: {
+          angular: 'angular',
+          jquery: '$',
```
Comment (antleblanc) [inline packages/components/ng-ovh-feature-flipping/rollup.config.js:22]:
```suggestion
```

File: packages/components/ng-ovh-feature-flipping/src/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import angular from 'angular';
+
+import provider from './provider';
+
+const moduleName = 'ngOvhFatureFlipping';
```
Comment (antleblanc) [inline packages/components/ng-ovh-feature-flipping/src/index.js:5]:
```suggestion
const moduleName = 'ngOvhFeatureFlipping';
```

File: packages/components/ng-ovh-feature-flipping/src/provider.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  constructor() {
+    this.applicationName = null;
+  }
+
+  /**
+   * Set the application name that will request the 2API feature availability.
+   *
+   * @param {string} applicationName The current application name
+   */
+  setApplicationName(applicationName) {
```
Comment (antleblanc) [inline packages/components/ng-ovh-feature-flipping/src/provider.js:13]: Couldn't be turned into a setter?

```suggestion
  set applicationName(applicationName) {
```

**Comment by FredericEspiau** [comment]
Do you have a document somewhere displaying how this works ? I've only seen 2 ou 3  PRs but no global project :)

**Comment by ovh-cds** [comment]
CDS Report installation#12360.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#12360.1 ✘  
* Stage 1
  * Install ✘

File: packages/manager/apps/telecom/src/components/sidebar/sidebar.config.js [hunk 6]
```diff
@@ -103,7 +104,11 @@ angular
         }
 
         // add sidebar SMS item
-        if (get(count, 'sms', beta)) {
+        if (
+          (get(count, 'sms') &&
+            featuresAvailabilities.isFeatureAvailable('sms')) ||
+          beta
+        ) {
```
Comment (cbourgois) [inline packages/manager/apps/telecom/src/components/sidebar/sidebar.config.js:111]:
```suggestion
        if (featuresAvailabilities.isFeatureAvailable('sms') && 
          (get(count, 'sms') || beta)
        ) {
```

I think we want to display "SMS" in the sidebar if we have sms services or if we are in beta and this and only if the feature is enabled.

Comment (Jisay) [reply packages/manager/apps/telecom/src/components/sidebar/sidebar.config.js]: Sounds good for me

File: packages/components/ng-at-internet/package.json [hunk 7]
```diff
@@ -42,4 +42,4 @@
   "peerDependencies": {
     "angular": "~1.6"
   }
-}
\ No newline at end of file
+}
```
Comment (marie-j) [inline packages/components/ng-at-internet/package.json:45]: Prettier ? Files should not be impacted in this PR though

Comment (Jisay) [reply packages/components/ng-at-internet/package.json:45]: It's since I am rebased on `develop` that these new lines appear

**Comment by antleblanc** [comment]
@Jisay Could you please resolve existing conflicts? In order to be able to merge it and also remove extra noise from the `yarn.lock` file that's not related to this Pull Request.

Thank you!

File: packages/components/ng-ovh-feature-flipping/package.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@ovh-ux/ng-ovh-feature-flipping",
+  "version": "1.0.0",
```
Comment (antleblanc) [inline packages/components/ng-ovh-feature-flipping/package.json:3]: Delegate it to the release script: 

```suggestion
  "version": "0.0.0",
```

File: packages/manager/apps/telecom/package.json [hunk 9]
```diff
@@ -43,6 +43,7 @@
     "@ovh-ux/ng-ovh-checkbox-table": "^2.0.0",
     "@ovh-ux/ng-ovh-contact": "^5.0.1",
     "@ovh-ux/ng-ovh-contracts": "^3.1.2",
+    "@ovh-ux/ng-ovh-feature-flipping": "1.0.0",
```
Comment (antleblanc) [inline packages/manager/apps/telecom/package.json:46]:
```suggestion
    "@ovh-ux/ng-ovh-feature-flipping": "^0.0.0 || ^1.0.0 ",
```

File: packages/manager/modules/sms/package.json [hunk 10]
```diff
@@ -28,6 +28,7 @@
     "@ovh-ux/ng-at-internet-ui-router-plugin": "^3.1.0",
     "@ovh-ux/ng-ovh-checkbox-table": "^2.0.0",
     "@ovh-ux/ng-ovh-contracts": "^3.1.2",
+    "@ovh-ux/ng-ovh-feature-flipping": "1.0.0",
```
Comment (antleblanc) [inline packages/manager/modules/sms/package.json:31]:
```suggestion
    "@ovh-ux/ng-ovh-feature-flipping": "^0.0.0 || ^1.0.0",
```

**Comment by antleblanc** [comment]
Got three approvals before rewritting git history.

It will be launched with the next release.

**Pull request merged**